astropy/units/format/unicode_format.py

```diff
@@ -33,6 +31,17 @@ class Unicode(console.Console):
     def _format_mantissa(cls, m):
         return m.replace("-", "−")
 
+    @classmethod
+    def _format_unit_power(cls, unit, power=1):
+        name = cls._get_unit_name(unit)
+        # Check for superscript units
+        if power != 1 and name in ("°", "e⁻", "″", "′", "ʰ"):
+            name = unit.short_names[0]
+            name += cls._format_superscript(utils.format_power(power))
```

Comment:
Generally, I like to punt to `super()` when possible, but here it feels a bit superfluous since it is not done for all cases and, as is, means that one has to get `name` twice. So, I think it is better to follow `latex` and make it more specific, as I suggested in my earlier comment, which, with your change to `name in ...` (better!) is,
```
name = cls._get_unit_name(unit)
if power != 1:
    if name in ("°", "e⁻", "″", "′", "ʰ"):
        name  = short_name
    name += cls._format_superscript(utils.format_power(power))
return name
```